mne/io/edf/edf.py

```diff
@@ -91,8 +135,14 @@ class RawEDF(BaseRaw):
 
     @verbose
     def __init__(self, input_fname, montage, eog=None, misc=None,
-                 stim_channel=-1, annot=None, annotmap=None, exclude=(),
+                 stim_channel=True, annot=None, annotmap=None, exclude=(),
                  preload=False, verbose=None):  # noqa: D102
+        if stim_channel is True:
+            warn("stim_channel will default to 'auto' in version 0.16. "
+                 "Set stim_channel explicitly to avoid this warning.",
+                 DeprecationWarning)
+            stim_channel = 'auto' if input_fname[-4:] == '.gdf' else -1
```

Comment:
The GDF event reading wasn't working in the last release anyway (fix within cycle), so I thought it's useless to make it  -1 for 0.15.